### ISSUE TITLE
[INFRA- Contributor's Github]  full access - Pilot

### ISSUE THREAD

**Issue by aeiras** [microprofile/microprofile-marketing#558]
From bugs Related to this fixed BUG Bug 483563 [https://bugs.eclipse.org/bugs/show_bug.cgi?id=483563]  - Allow assignment of Github issues to contributors & Bug 562071 [https://bugs.eclipse.org/bugs/show_bug.cgi?id=562071] - Allow Github contributors to have "triage" rights

MPWG Contributors to be added as Contributors in the MP repo as follows from the Marketing side:  https://docs.google.com/spreadsheets/d/1ntRFZiKRvZpXSvfHM57mIDXPdtylDPyYiswegMkKKyw/edit#gid=0

Ed @edbratt has volunteered to help pilot the feature as a contributor. :)  

@chrisguindon hola!!!

I am most excited to finally put into use the hard work from the 8+ months we spent on fixing the +4yrs old bug. 

Can you please direct us on how to make of use the feature?  I hope that it can even more scalable and better sustainable after the time that has passed. 
The blog we drafted likely needs updating so pulling you in as that was Q42019.  
Please notice that I created the contributors' list and used only their github ID. Do we need their email associated with the signature of the ECA as well or only? 

In advance, thank you Ed and Chris!
Once we tackled MP git, we will also apply the feature to improving the github dynamics in the Jakarta EE space :)

**Comment by chrisguindon** [comment]
Hi @aeiras,

We have documentation about this with our Github sync script:
https://github.com/EclipseFdn/eclipsefdn-github-sync#gaining-access-to-a-project

tldr; Contributors must be added by committers/pl in their project form on projects.eclipse.org. Once that's done, our sync script will add or remove new contributors for each of our projects.

**Comment by aeiras** [comment]
Chris, 

Thank you for sharing the updated wiki, you are amazing! 
I will start the test with Ed and follow the steps sometime this week. :)

**Comment by aeiras** [comment]
@@edbratt, 
What is your email associated with your EF github? 
Please see the sheet below. Once I have it, without rush, we can start the pilot. Thank you :) 

MPWG Contributors to be added as Contributors in the MP repo as follows from the Marketing side: https://docs.google.com/spreadsheets/d/1ntRFZiKRvZpXSvfHM57mIDXPdtylDPyYiswegMkKKyw/edit#gid=0

**Comment by edbratt** [comment]
ed dot bratt at oracle dot com (Sorry, I thought I had sent this earlier)

**Comment by aeiras** [comment]
Access sent, thank you Ed.

**Comment by aeiras** [comment]
Hola beautiful @chrisguindon,

Yesterday, Ed gained access via email to become a Contributor in MP. 
As of today, it has been 24hrs and nothing has changed. 
Either the system is not updating or we are missing a step not listed in the Guidelines. 

Can you please unblock us? 

Thank you,

**Comment by chrisguindon** [comment]
Hi @aeiras,

I can see in our GitHub logs that @edbratt was added to the contributors' team yesterday:

>  Added edbratt to the eclipse/technology-microprofile-contributors team
> Canada 22 hours ago


A contributor will have the "triage" permission level on the associated project git repositories.  The triage permission level is documented here: https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-permission-levels-for-an-organization

**Comment by aeiras** [comment]
Thank you for confirming what you see Chris. 

Ok, so there is no notification from github.

Very cool. based on your update, Ed gained immediate access, within 2hrs, to triage access to MP repos. 

 @edbratt, it seems you are all set to test github. What do you see? Can you confirm your triage capabilities in MP? 

Thank you so much Chris for prompt response, 

PS: I owe you the publication of that blog, this will populate it as well. Undertaking the feature, finally. It is soooo EXCITING!

**Comment by chrisguindon** [comment]
@aeiras No problem! The email that Ed received was from github. It was probably his invitation to join the Eclipse organization. 

The user will gain immediate access to the triage access once the email invitation is accepted. The email is only sent once PER organization on github.

**Comment by aeiras** [comment]
I see. 

@edbratt hasn't received an email yet. 
Ed, that email might be in your spam...   you must accept the access to gain triage. 

Very cool, Chris! :)

**Comment by edbratt** [comment]
Ah. I was looking for something I didn't have. This provides "Triage" 
access -- (as opposed to Write access). That's semi-defined here 
<https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-permission-levels-for-an-organization>. 
It doesn't allow a contributor to actually merge anything into a 
repository -- that's still reserved for elected committers. These 
members can work with issues and some managerial aspects of pull requests.

If that's what was expected, that appears to be what I have been granted.

Thank you Chris for pointing me to the e-mail, which in fact was a 
confirmation that I'd been added to 
'technology-microprofile-contributors' GitHub team which provides 
'triage' role for all MicroProfile repositories.

With all the messages I get, I had missed that one and, I had thought (I 
guess mistakenly) I would be allowed write permission for (at least 
some) repositories. The subject was: "Eclipse Webmaster team added you 
to Eclipse Foundation team technology-microprofile-contributors" from 
"Eclipse Foundation" (github user).

Though, in my case there was nothing for me to accept since I'm already 
a committer to other Eclipse repositories -- perhaps others (who are not 
yet committers) will get an additional e-mail that requires a confirmation.

Amelia, if this is what you were expecting, I can confirm it works.

-- Ed

**Comment by aeiras** [comment]
Yes, Ed! 

This feature is what Chris made possible for us (8months of bug exchanges) in 2019. 

Chris, you know how much I appreciate you. Thank you for everything you do to make the EF ecosystem better for #ossDOERS!

Ed, 

I will close this issue so that Chris is not distracted by what comes next. Onboarding the other Mktg contributors listed in the spreadsheet. 
We need to document how quickly.  You already documented core information internally, so I wonder if you can own that draft document (git issue needed for the task) using the MP community public documentation folder https://drive.google.com/drive/u/0/folders/13qmKV0inSuxX3uhzpD_mbLKtmgmHI2e4

I will help with screenshots, etc., to show Committers what happens when the email gets added & then linked to the contributor's GitHub, etc.  We want all MP committers to enable their most active contributors to use the feature. :) Amazing!
